android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -122,10 +125,83 @@ class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
         }
     }
 
+    @ReactMethod
+    fun upload(baseUrl: String, endpoint: String, file: String, taskId: String, options: ReadableMap?, promise: Promise) {
+        val body: RequestBody;
+        val uri = Uri.parse(file);
+
+        // check for multi-parts options
+        if(options != null && options.hasKey("multipart")){
+            val multipartOptions = options.getMap("multipart")!!
+
+            val multipartBody = MultipartBody.Builder();
+
+            if(multipartOptions.hasKey("fileKey") && multipartOptions.getString("fileKey") !== ""){
+                multipartBody.addFormDataPart(multipartOptions.getString("fileKey")!!, uri.lastPathSegment, UploadFileRequestBody(reactApplicationContext, uri, 0, ProgressListener(reactApplicationContext, taskId)))
+            } else {
+                multipartBody.addPart(UploadFileRequestBody(reactApplicationContext, uri, 0, ProgressListener(reactApplicationContext, taskId)))
```

Comment:
I thought we decided to use "files" as the default key

Reply:
Ah whoops - will change.